plugins/modules/ovirt_qos.py

```
@@ -0,0 +1,264 @@
+            return otypes.QosType.HOSTNETWORK
+        elif type == 'cpu':
+            return otypes.QosType.CPU
+        else:
+            return None
+
+    def build_entity(self):
+        """
+        Abstract method from BaseModule called from create() and remove()
+
+        Builds the QoS from the given params
+
+        :return: otypes.QoS
+        """
+        return otypes.Qos(
+            name=self._module.params.get('id') if self._module.params.get('id') else self._module.params.get('name'),
+            type=self._get_qos_type(self._module.params.get('type')),
+            description=self._module.params.get('description') if self._module.params.get('description') else None,
+            max_iops=int(self._module.params.get('max_iops')) if self._module.params.get('max_iops') is not None else None,
+            max_read_iops=int(self._module.params.get('read_iops')) if self._module.params.get('read_iops') is not None else None,
+            max_read_throughput=int(self._module.params.get('read_throughput')) if self._module.params.get('read_throughput') is not None else None,
+            max_throughput=int(self._module.params.get('max_throughput')) if self._module.params.get('max_throughput') is not None else None,
+            max_write_iops=int(self._module.params.get('write_iops')) if self._module.params.get('write_iops') is not None else None,
+            max_write_throughput=int(self._module.params.get('write_throughput')) if self._module.params.get('write_throughput') is not None else None,
+        )
+
+def _get_qoss_service(connection, dc_name):
+    """
+    Gets the qoss_service from the data_center provided
+
+    :returns: ovirt.services.QossService or None
+    """
+    dcs_service = connection.system_service().data_centers_service()
+    dc = search_by_name(dcs_service, dc_name)
+
+    if dc is None:
+        dc = get_entity(dcs_service.service(dc_name))
+        if dc is None:
+            return None
```

Please raise the exception that the Datacenter was not found.